server/src/controllers/AuthController.js

```diff
@@ -77,12 +77,12 @@ module.exports = {
         expiresIn: 60 * 60,
       });
 
-      res.status(200).json({
+      return res.status(200).json({
         username: user.username,
         token: token,
```

Comment:
You also need to return the JSON object for the register function on line 41.

Reply:
Good catch, thanks! Will update the same in the next commit

Reply:
About this, this will probably get fixed in #35 . So, let's focus on getting the front end right in this PR.